app/models/piece.rb

```diff
@@ -71,11 +69,30 @@ def move_legal?(to_row, to_col)
     end
   end
 
-  def move_destination_obstructed?(_to_row, _to_col)
-    false
-  end
+  def move_obstructed?(to_row, to_col)
+    Rails.logger.debug "***** Move_obstructed from #{row},#{col} to #{to_row},#{to_col}"
+    if type == "Knight"
+      Rails.logger.debug "KNIGHT. cannot obstruct move"
+      false
+    end
+
+    obstruction_array = []
+    if row == to_row
```

Comment:
I will consider this; however, I am currently working on using the board.grid array. So the final solution will be completely different from what we see now.